### PR TITLE
Add AlbertForPreTraining and TFAlbertForPreTraining models.

### PR DESCRIPTION
**Pull request by jarednielsen**
I mirrored the structure of `BertForPreTraining` and `TFBertForPreTraining`.

A few differences between the PyTorch and TensorFlow implementations of Bert I noticed along the way:
- TFBertForSequenceClassification has dropout in the classifier head, but TFBertForPreTraining does not. I included dropout in the TFAlbertForPreTraining classifier head.
- TFBertForPreTraining does not include an option to calculate the loss, but BertForPreTraining does. Fixing this will require more wrangling of the dataset and preprocessor, so I'll save that for a later PR. Mirrored the structure here.
- BertForPreTraining has attributes: `self.bert`, `self.cls.predictions`, `self.cls.seq_relationship`. TFBertForPreTraining has attributes: `self.bert`, `self.dropout`, `self.classifier`. I chose to unify these attributes in AlbertForPreTraining and TFAlbertForPreTraining under `self.albert`, `self.cls.predictions`, and `self.cls.sop_classifier`.
- TFAlbertMLMHead adds two biases to the final output: `self.decoder_bias` and `self.bias`. TFBertMLMHead only adds one bias: `self.bias`.

### CODE REVIEW

**Comment by LysandreJik** [comment]
Hi! This is really cool, thanks for taking the time to do that! A few notes on your remarks:

> TFBertForPreTraining does not include an option to calculate the loss, but BertForPreTraining does. 

This is not a bug, it's done on purpose. We try to respect as much as possible each library's paradigm. With PyTorch you can compute the loss both inside and outside the model, so we make it easy for the user to supply the labels and compute a standard loss.

For TensorFlow however, the loss is usually computed outside the model, especially with Keras. None of our TensorFlow models computes the loss for that very reason.

> TFAlbertMLMHead adds two biases to the final output: self.decoder_bias and self.bias. TFBertMLMHead only adds one bias: self.bias.

That's, unfortunately, a mistake on my side. https://github.com/huggingface/transformers/pull/4076


I'm taking a look at your PR, and will see if we can upload weights on S3 containing the SOP weights as well.

**Comment by jarednielsen** [comment]
Hi @LysandreJik, any updates on your side? I know last week was ICLR. We're prepping a code release and blog post, so it would be great to have this merged in soon :)

**Comment by LysandreJik** [comment]
Hi @jarednielsen, I took a look today, and it's in good shape! There's a few things to update, however, as this is an opportunity to update our checkpoints to include the SOP.

In order to do this, I had to do a few changes, especially removing the `AlbertPreTrainingHeads` class so that the masked lm layer (previously `AlbertForPreTraining.cls.predictions`) could be at the same level than the `AlbertForMaskedLM` class: `AlbertForMaskedLM.predictions`.

There's a few other changes that need to be done in the conversion utility. I'll look into doing TensorFlow tomorrow and will try merge tomorrow evening + upload the updated checkpoints on the S3; is this timeline okay with you?

Do you mind if I push directly on your fork, so that all the changes are visible in this PR?

**Comment by jarednielsen** [comment]
Thanks for the prompt response! Yes, that timeline sounds great; feel free to push directly onto my fork.

**Comment by LysandreJik** [comment]
Cool, will do that. We're releasing transformers `v2.9` today so it won't be in it, but it will be in the next version which should be released sometimes next week.

**Comment by LysandreJik** [comment]
Thanks a lot for your contribution! Will upload the weights on S3 in the coming days.
